source/keyboardHandler.py

```diff
@@ -275,8 +281,11 @@ def internal_keyUpEvent(vkCode,scanCode,extended,injected):
 
 #Register internal key press event with  operating system
 
-def initialize():
+
+def initialize(watchdogObserver=None):

```

Comment:
This being an optional parameter doesn't follow the intend of the watchdog observer being absolutely required within the keyboard handler. I don't think backwards compatibility has to be taken into account here. However, if it is decided otherwise, all checks for the property on the observer must first check whether there's an observer assigned to the variable.

```suggestion
def initialize(watchdogObserver):
```